### ISSUE TITLE
if a player get hit w/ spell they can't damage mobs

### ISSUE THREAD

**Issue by tensiyoka** [elBukkit/MagicPlugin#1112]
Have a reoccurring bug on my server where sometimes when a player gets hit with a Spell  (offensive or defensive)  they can no longer damage mobs w/ attacks of any type magic or otherwise.  can't get it too replicate on any consistent basis.  Just a sorta random thing.  If the player logs out and back in...  seems to fix it.

**Comment by SDavey149** [comment]
Having the same problem on Minecraft Paper server. Doesn't always happen but it does happen quite often, have to disconnect and rejoin to be able to damage mobs again.

**Comment by LordBarl** [comment]
Also having that issue, and hard to avoid, there are too many dragons. Once it happens, no mob takes damage. Arrows bounce. Melee weapons do nothing. Spells do nothing. Not piling on, but breaking combat is a problem.

**Comment by cosmic80** [comment]
yeah i've encountered this issue too. I noticed it when casting cure, after that i noticed i couldn't hurt anything anymore, left and rejoined, tested it one more time to replicate and sure enough - as soon as i cast cure no mobs will take damage from me.

**Comment by NathanWolf** [comment]
Thanks for the specific pointers- I have not experienced nor heard of this problem before this thread, so I'll be really surprised if there isn't some other ingredient, in particular an anti-cheat plugin.

I'll test out cure and let you know.

If you think there are too many dragons, the spawn rules are editable:

`/meditor world overworld_mobs.replaceflying` should be pretty straightforward. You may need to do the same thing for `nether_mobs` and `end_mobs`. But please keep issues on-topic, if you need more help with mob spawning open another ticket.

**Comment by cosmic80** [comment]
For sure, if I can be helpful with anything else lmk and I'll report on it. I think you're probably right that there is some other ingredient at play. I replicated it with the heal spell too which really sucks lol, i was so looking forward to using that.

I can say for certain that I'm not really using much in the way of anticheat - this is a small private dedicated server i run off my own hardware for family and friends.

I do sort of recall it working fine before, then I added craftbook, worldedit/guard + extra flags, protocolLib, LibsDisguises & morph, jobs & medieval factions before I noticed it. So I'll probably experiment with disabling some of these and if I can find the culprit I'll come back so others know what causes it.

**Comment by Remodactyl** [comment]
I have the same issue where this happens very rarily. I have the following plugins, I put the ones shared with cosmic80 on top:
* WorldEdit 
* WorldGuard
* LibsDisguises
* ProtocolLib
* GriefPrevention
* XEssentials
* ViaVersion
* ViaBackwards
* Multiverse-Core
* LucksPerms
* VoxelSniper
* CustomCrafting
* WolfyUtilities
* CommandAPI

I only expect some of the top ones to fiddle with this stuff, but just in case.

**Comment by NathanWolf** [comment]
To all of you who are having this problem regularly, I apologize! I can imagine it's frustrating, I wish I had some idea offhand of what it could be, but I will really try to do some in-depth debugging as soon as I can.

For anyone who has the problem pretty reliably, could you please try using the command `/mage debug 200` ?

That should print to your chat anytime you damage something, and I *think* may print why damage is prevented, if it is coming from Magic.  I appreciate any help you can give tracking this down, thank you!

**Comment by NathanWolf** [comment]
I found and reproduced this issue. Thanks to everyone who helped me track this down!

It was caused by a change introduced in 10.8.2, and will be fixed in the next dev build.
